Glioblastoma example notebook

njmei opened the pull request:
This PR is intended to address:
https://github.com/AllenInstitute/AllenSDK/issues/1648

Handy link to view the notebook itself:
https://app.reviewnb.com/AllenInstitute/open_dataset_tools/pull/3/

njmei committed:
Add proper .gitignore for Python files

njmei committed:
Add example notebook for accessing Ivy GAP data

njmei committed:
Update repository README.md

wbwakeman commented on Programmatic_access_of_Ivy_Glioblastoma_Atlas_Project_data.ipynb:
```diff
@@ -0,0 +1,4225 @@
+    "- Gene expression masks for all ISH images\n",
+    "- Annotated mask images of tumor structures for all ISH and H&E images\n",
+    "- RNA sequencing data for 270 samples from 44 tumors\n",
+    "- Companion clinical database hosted at https://ivygap.swedish.org (registration required)\n",
```
URL should be: https://ivygap.org/

wbwakeman commented on Programmatic_access_of_Ivy_Glioblastoma_Atlas_Project_data.ipynb:
```diff
@@ -0,0 +1,4225 @@
+   "source": [
+    "## Accessing Ivy Glioblastoma Atlas Project data with AWS S3\n",
+    "\n",
+    "The image data for the project is being made available as an Amazon Web Services (AWS) public dataset to enable computational scientists easy access to a rich, well-annotated data set for training and validation of machine learning, classification, and developing computer vision applications such as image segmentation. The data are free and publicly available from the AWS Simple Storage Serv ...
```
I'm still pretty confused by what access people should expect.  I don't think this console link works for non-admins, even if they are logged in with their AWS account.  Probably should remove that URL.  The S3 ARN should stay though.

wbwakeman commented on Programmatic_access_of_Ivy_Glioblastoma_Atlas_Project_data.ipynb:
```diff
@@ -0,0 +1,4225 @@
+    "The root of the S3 bucket contains 3 metadata \\*.json files (`donor_metadata.json`, `specimen_metadata.json`, and `section_metadata.json`) that can be used to explore the dataset and determine download links. \n",
+    "\n",
+    "The relationship between donors, specimens, and sections is as follows:\n",
+    "- Each donor can have one or more (tumor tissue) specimens\n",
```
I like this clear description.  One addition:
"Each donor (a human cancer patient) can have one or more (tumor tissue) specimens"

wbwakeman commented on Programmatic_access_of_Ivy_Glioblastoma_Atlas_Project_data.ipynb:
```diff
@@ -0,0 +1,4225 @@
+   "source": [
+    "## Exploring section metadata and interacting with images\n",
+    "\n",
+    "The section metadata table includes information about sliced tissues sections obtained from specimens including primer sequence data used for in situ hybridization experiments.\n",
```
sliced tissues sections -> sliced tissue sections

wbwakeman commented on Programmatic_access_of_Ivy_Glioblastoma_Atlas_Project_data.ipynb:
```diff
@@ -0,0 +1,4225 @@
+   "source": [
+    "The section image table contains data as well as metadata for sub-images within a section. Images are loaded in as instances of the Python Image Library (PIL) `Image` class. For more details about this class, please see: https://pillow.readthedocs.io/en/stable/reference/Image.html\n",
+    "\n",
+    "For the purposes of this demo notebook, it is sufficient to know that one can use matplotlib plot some previews of `Image` objects:"
```
use matplotlib plot -> use matplotlib to plot

wbwakeman commented on Programmatic_access_of_Ivy_Glioblastoma_Atlas_Project_data.ipynb:
```diff
@@ -0,0 +1,4225 @@
+    }
+   ],
+   "source": [
+    "section_image_table = section_image_loader(\n",
```
Oh man. I got a "MemoryError" while it was downloading the 10th image of the set. Not sure if there is anything to be done about it.  Unfortunately, it does prevent me from going forward with the notebook

njmei replied:
How much RAM does the system you tried this on have?

danielsf reviewed (changes requested):
What is the rationale for having the utility functions (`section_image_loader`, `section_metadata_cached_loader`, `section_image_dowloader`) defined in the notebook, rather than in a module that the notebook imports?

If we put the functions in a module then people who (like me) prefer working in scripts to working in notebooks can import the functions from their own scripts. I suspect this would be especially useful for someone who wants to start a job downloading the whole dataset and walk away from it while it runs.

I'm open to arguments why the notebook should remain as-is, but that is my opinion.

djkapner commented:
I added come ReviewNB comments.
Why are these files in the root of this repo?

danielsf reviewed (approved):
Based on our conversation after stand up, this is ready to merge.

njmei committed:
Address PR recommendations

- Adds docstrings
- Small documentation changes
- Splits section_image_loader into s3_section_image_loader
  and local_section_image_loader

njmei commented:
@djkapner Okay, I think I've addressed most of your comments. Let me know if anything else stands out.

wbwakeman reviewed (commented):
submitting these now.  Maybe they were not seen before?
